File: tools/token_bench/programs/06_word_frequency/word_freq.py
from collections import Counter

text = "the quick brown fox jumps over the lazy dog the fox the dog and the cat sat on the mat the quick brown fox"

words = text.split()
print(f"Total: {len(words)}")
print(f"Unique: {len(set(words))}")

counts = Counter(words)
for word, count in counts.items():
    print(f"{word}: {count}")

longest = max(words, key=len)
print(f"Longest: {longest} ({len(longest)} chars)")

avg = sum(len(w) for w in words) / len(words)
print(f"Avg length: {avg}")

File: tools/token_bench/programs/06_word_frequency/word_freq.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	text := "the quick brown fox jumps over the lazy dog the fox the dog and the cat sat on the mat the quick brown fox"

	words := strings.Fields(text)
	fmt.Printf("Total: %d\n", len(words))

	unique := make(map[string]bool)
	for _, w := range words {
		unique[w] = true
	}
	fmt.Printf("Unique: %d\n", len(unique))

	counts := make(map[string]int)
	for _, w := range words {
		counts[w]++
	}
	for word, count := range counts {
		fmt.Printf("%s: %d\n", word, count)
	}

	longest := words[0]
	for _, w := range words {
		if len(w) > len(longest) {
			longest = w
		}
	}
	fmt.Printf("Longest: %s (%d chars)\n", longest, len(longest))

	totalLen := 0
	for _, w := range words {
		totalLen += len(w)
	}
	avg := float64(totalLen) / float64(len(words))
	fmt.Printf("Avg length: %f\n", avg)
}
